From de62879076935bfb07eb52b23b25b5716d514df0 Mon Sep 17 00:00:00 2001
From: Brian Wood <brian.j.wood@intel.com>
Date: Wed, 17 Sep 2014 15:57:34 -0700
Subject: [PATCH 2/2] Updated the BCMDHD driver for 3.14 API change

This commit updated for the changes from the 3.10
PCI Exported function to what is available in 3.14.

Signed-off-by: Brian Wood <brian.j.wood@intel.com>
---
 drivers/net/wireless/bcmdhd/dhd_pcie_linux.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/net/wireless/bcmdhd/dhd_pcie_linux.c b/drivers/net/wireless/bcmdhd/dhd_pcie_linux.c
index 781c17c0359c..3a01f913df02 100644
--- a/drivers/net/wireless/bcmdhd/dhd_pcie_linux.c
+++ b/drivers/net/wireless/bcmdhd/dhd_pcie_linux.c
@@ -732,11 +732,16 @@ dhdpcie_enable_device(dhd_bus_t *bus)
 	if(pch == NULL)
 		return BCME_ERROR;
 
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 0, 0))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 0, 0) && LINUX_VERSION_CODE < KERNEL_VERSION(3, 14, 0))
 	if (pci_load_saved_state(bus->dev, pch->state))
 		pci_disable_device(bus->dev);
 	else {
 #endif /* LINUX_VERSION_CODE >= KERNEL_VERSION(3, 0, 0) */
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 14, 0))
+	if (pci_load_and_free_saved_state(bus->dev, &pch->state))
+		pci_disable_device(bus->dev);
+	else {
+#endif /* LINUX_VERSION_CODE >= KERNEL_VERSION(3, 14, 0) */
 		pci_restore_state(bus->dev);
 		ret = pci_enable_device(bus->dev);
 		if(!ret)
-- 
2.0.3

